packages/@uppy/provider-views/src/SharedHandler.js

```diff
@@ -42,6 +44,20 @@ module.exports = class SharedHandler {
       } else {
         currentSelection = items.slice(currentIndex, prevIndex + 1)
       }
+      // Check restrictions on each file in currentSelection,
+      // reduce it to only contain files that pass restrictions
+      currentSelection = currentSelection.reduce((reducedCurrentSelection, item) => {
+        const uppy = this.plugin.uppy
+        const validatedRestrictions = uppy.validateRestrictions(
+          remoteFileObjToLocal(item),
+          [...uppy.getFiles(), ...reducedCurrentSelection]
+        )
+        if (!validatedRestrictions.result) {
+          uppy.info({ message: validatedRestrictions.reason }, 'error', 5000)
+          return reducedCurrentSelection
+        }
+        return [...reducedCurrentSelection, item]
+      })
```

Comment:
@lakesare could you take a look at the reducer here? makes sense? thx!

Reply:
Looks right, but makes me worried about the initial value of `reducedCurrentSelection` e.g..
I'd go for a more explicit:

```
const validSelection = []
currentSelection.forEach((file) => {
  const validatedRestrictions  = uppy.validateRestrictions(
     remoteFileObjToLocal(file),
     [...uppy.getFiles(), ...validSelection]
  )
  if (!validatedRestrictions.result) {
    uppy.info({ message: validatedRestrictions.reason }, 'error', 5000)
  } else {
    validSelection.push(file)
  }
})

this.plugin.setPluginState({ currentSelection: validSelection })
```